splitio/storage/pluggable.py

```diff
@@ -0,0 +1,195 @@
+        :type split_name: str
+
+        :return: True if the split was found and removed. False otherwise.
+        :rtype: bool
+        """
+        split = self.get(split_name)
+        if not split:
+            _LOGGER.warning("Tried to delete nonexistant split %s. Skipping", split_name)
+            return False
+
+        self._pluggable_adapter.delete(self._prefix + split_name)
+        self._pluggable_adapter.decrement(self._traffic_type_prefix + split.traffic_type_name, 1)
+        if self._pluggable_adapter.get(self._traffic_type_prefix + split.traffic_type_name) == 0:
+            self._pluggable_adapter.delete(self._traffic_type_prefix + split.traffic_type_name)
+        return True
+
+    def get_change_number(self):
+        """
+        Retrieve latest split change number.
+
+        :rtype: int
+        """
+        return self._pluggable_adapter.get(self._split_till_prefix)
+
+    def set_change_number(self, new_change_number):
+        """
+        Set the latest change number.
+
+        :param new_change_number: New change number.
+        :type new_change_number: int
+        """
+        self._pluggable_adapter.set(self._split_till_prefix, new_change_number)
+
+    def get_split_names(self):
+        """
+        Retrieve a list of all split names.
+
+        :return: List of split names.
+        :rtype: list(str)
+        """
+        return [split.name for split in self.get_all()]
+
+
+    def get_all(self):
+        """
+        Return all the splits.
+
+        :return: List of all the splits.
+        :rtype: list
+        """
+        return [splits.from_raw(split) for split in self._pluggable_adapter.get_keys_by_prefix(self._prefix)]
+
+    def traffic_type_exists(self, traffic_type_name):
+        """
+        Return whether the traffic type exists in at least one split in cache.
+
+        :param traffic_type_name: Traffic type to validate.
+        :type traffic_type_name: str
+
+        :return: True if the traffic type is valid. False otherwise.
+        :rtype: bool
+        """
+        return self._pluggable_adapter.get(self._traffic_type_prefix + traffic_type_name) != None
+
+    def kill_locally(self, split_name, default_treatment, change_number):
+        """
+        Local kill for split
+
+        :param split_name: name of the split to perform kill
+        :type split_name: str
+        :param default_treatment: name of the default treatment to return
+        :type default_treatment: str
+        :param change_number: change_number
+        :type change_number: int
+        """
+        split = self.get(split_name)
+        if not split:
+            return
+        if self.get_change_number() > change_number:
+            return
+        split.local_kill(default_treatment, change_number)
+        self._pluggable_adapter.set(self._prefix + split_name, split.to_json())
+        self.set_change_number(change_number)
+
+    def increase_traffic_type_count(self, traffic_type_name):
```

Comment:
I think you can remove `increase_traffic_type_count` and `decrease_traffic_type_count` methods, since `put` and `remove` are already incrementing and decrementing the traffic type. Or you can use them in `put` and `remove` respectively.

Reply:
As a reference, you can check what we do in JS https://github.com/splitio/javascript-commons/blob/development/src/storages/pluggable/SplitsCachePluggable.ts#L31

Reply:
that is correct, I will reference them from put and remove, since they are in the spec